### ISSUE TITLE
Actually i have a lot of questions so i will go one by one)

### ISSUE THREAD

**Issue by Nazarko12345** [learnml-today/object-detection-with-pytorch#3]
1 task.
I think that denoising of image solves using Unets(I have writen simple one with fews layers).
-What types of upsampling u recommend to use? 
-What types of data augmentation are most popular to use in them?Do u use Laplacian pyramids?
-Also because Unets Fully Convolutional can we train them for different size images(group them to have same sizes in batch)
-Please give few papers about them which you recommend to read(or about segmentation task)
2 task.
-Segmentation is pixel wise classification but why u use MSELoss for mask.
-How to combine classification/segmentation/detection?Using 3 seperate neural nets from resnet and optimize 3 different losses?
-How much effect resizing of images impact on our quality of network(For example we resizing 20 images with tiger do we will have a problem with dont resized tiger? or with real images?)
Sorry for such a long list)

**Comment by creotiv** [comment]
**I think that denoising of image solves using Unets(I have writen simple one with fews layers).**
So as for denoising look at this task as the way of optimizing the image by removing additional information, as GIF or JPG does.  Best fit for that any algorithm or NN that give the ability to minimize data. For example, u can use SVD, PCA, AutoEncoders, etc for that task.
As for Unet is not a good thing, because it has resid connections between layers, which is good for segmentation but not for compression.

**-What types of upsampling u recommend to use?**
Heh)) If you search papers on arxiv on that thing, you can find many articles with different thoughts even on one task. The problem is that hard to say definitely.
If you work with any image reconstruction algorithm u need to avoid all things that can change the data like ConvTranspose wich adding high checkboard artifcts. So the best fit here will be an interpolation and maybe PixelShufle(but last SR GAN was using interpolation).

**-What types of data augmentation are most popular to use in them?Do u use Laplacian pyramids?**
It depends on the task that I need to do. In most cases, there is no need to apply ant filters that can be replaced by Conv.
There are many types of augmentations like weather conditions, image sizes, image params(lightness, contrast, etc). Do you need to add them? this depends on how the model should work and in which conditions. For example, if image always same size and scale I don't need to do multiscale augmentation.
So don't do work that don't need, and always understand that + 0.0001 to accuracy doesn't cost 3 months of work

**-Also because Unets Fully Convolutional can we train them for different size images(group them to have same sizes in batch)**
1) u can scale different shapes of images to one shape like 300x300 and make a batch
2) u can pad smaller images with zeros and include in batch (this done in FasterRcnn that we will train later)
3) comes up with something new)

**-Please give few papers about them which you recommend to read(or about segmentation task)
2 task.**
It's hard to understand area by a few papers, basically, you need to read them all to understand why some things were replaced by another. The current state of the art is the MaskRCNN https://arxiv.org/abs/1703.06870 (people also using cascades on them, but that basically same network repeated)
Also you can look here: https://paperswithcode.com/sota/instance-segmentation-on-coco

**-Segmentation is pixel wise classification but why u use MSELoss for mask.**
But our task was just to get bbox and not precise edge segmented image. and MSE giving better speed (in current case). But as for segmentation task you are right.

**-How to combine classification/segmentation/detection?Using 3 seperate neural nets from resnet and optimize 3 different losses?**
The task of backbone network(base) like resnet vgg is not classification, segmentation, etc their task is feature extraction. And it doesn't differs for classification, segmentation, detection. So u can use one network with 3 heads and 3 losses for each of them.
Also, multi-task networks have better results, that's why MaskRcnn so good.

**-How much effect resizing of images impact on our quality of network(For example we resizing 20 images with tiger do we will have a problem with dont resized tiger? or with real images?)
Sorry for such a long list)**
If trained data differs from used for inference - you 100% will have problem. Because CNN is not scale/rotation/warp invariant. they detect what they was teached, and that's all.
If you interested in invarinat networks i can recommend to read about Capsule Networks
https://medium.com/ai%C2%B3-theory-practice-business/understanding-hintons-capsule-networks-part-i-intuition-b4b559d1159b

Thanks for your question))

**Comment by Nazarko12345** [comment]
Thanks for answering.I have trained for single object detection with few classes for 15 iteration i get pretty high accuracy(0.9) and IoU for boxes (0.6).But with masks i have problem.I have 200 classes so i get output 201 (201x50x50) (+1 for background 50x50 mask size).So questions next:
 - How to find IoU for masks?(We have 201 output and dont know for what class to find? maybe try for first channel (background)?) 
- After training for 15 epochs i have a great background detection but for classes i get very bad result should i go on training?  
- Also should i create new topic for answering or its comfortable for you to ask here?

**Comment by creotiv** [comment]
Create new topic please
